lmcache/v1/storage_backend/cache_policy/lru.py

```diff
@@ -18,22 +20,38 @@ class LRUCachePolicy(BaseCachePolicy[OrderedDict[CacheEngineKey, Any]]):
 
     def __init__(self):
         logger.info("Initializing LRUCachePolicy")
+        self.chunk_hash_to_init_timestamp: Dict[Any, Any] = {}
+        self.stats_monitor = LMCStatsMonitor.GetOrCreate()
+        self.max_num_chunk_hash = 12500000
 
     def init_mutable_mapping(self) -> OrderedDict[CacheEngineKey, Any]:
         return OrderedDict()
 
+    def update_chunk_hash_dict(self, key: CacheEngineKey) -> None:
+        if key.chunk_hash not in self.chunk_hash_to_init_timestamp:
+            if len(self.chunk_hash_to_init_timestamp) >= self.max_num_chunk_hash:
+                # Clear the dictionary to avoid memory leak
+                self.chunk_hash_to_init_timestamp = {}
+            self.chunk_hash_to_init_timestamp[key.chunk_hash] = time.time()
+        if key.chunk_hash in self.chunk_hash_to_init_timestamp:
+            time_interval = (
+                time.time() - self.chunk_hash_to_init_timestamp[key.chunk_hash]
+            )
+            self.stats_monitor.on_chunk_reuse(time_interval)
```

Comment:
We can reduce the number of lookups in this function. Could be something like:
```suggestion
        curr_time = time.time()
        key_hash = key.chunk_hash
        if init_timestamp := self.chunk_hash_to_init_timestamp.get(key_hash, None):
            time_interval = curr_time - init_timestamp
            self.stats_monitor.on_chunk_reuse(time_interval)
        else:
            if len(self.chunk_hash_to_init_timestamp) >= self.max_num_chunk_hash:
                self.chunk_hash_to_init_timestamp.clear()
            self.chunk_hash_to_init_timestamp[key_hash] = curr_time
```